ts/doc.go

```diff
@@ -0,0 +1,139 @@
+
+With this in mind, Cockroach downsamples data before storing it; the original
+timestamp for each data point in a series is not recorded. Cockroach instead
+divides time into contiguous slots of uniform length (e.g. 10 seconds); if
+multiple data points for a series fall in the same slot, they are aggregated
+together.  Aggregated data for a slot records the count, sum, min and max of
+data points that were found in that slot. A single slot is referred to as a
+"sample", and the length of the slot is known as the "sample duration".
+
+Cockroach also uses its own key space efficiently by storing data for
+multiple samples in a single key. This is done by again dividing time into
+contiguous slots, but with a longer duration; this is known as the "key
+duration". For example, cockroach samples much of its internal data at a
+resolution of 10 seconds, but stores it with a "key duration" of 1 hour, meaning
+that all samples that fall in the same hour are stored at the same key. This
+strategy helps reduce the number of keys scanned during a query.
+
+Finally, cockroach can record the same series at multiple sample durations, in a
+process commonly known as a "rollup".  For example, a single series may be
+recorded with a sample size of 10 seconds, but also record the same data with a
+sample size of 1 hour. The 1 hour data will have much less information, but can
+be queried much faster; this is very useful when querying a series over a very
+long period of time (e.g. an entire month or year).
+
+A specific sample duration in cockroach is known as a Resolution. Cockroach
+supports a fixed set of Resolutions; each Resolution has a fixed sample duration
+and a key duration. For example, the resolution "Resolution10s" has a sample
+duration of 10 seconds and a key duration of 1 hour.
+
+###Source Keys
+
+Another dimension of time series queries is the aggregation of multiple series;
+for example, you may want to query the same data point across multiple machines
+on a cluster.
+
+While Cockroach will support this, in some cases queries are almost *always* an
+aggregate of multiple source series; for example, we often want to query storage
+data for a Cockroach accounting prefix, but data is always collected from
+multiple stores; the information on a specific range is rarely relevant, as they
+can arbitrarily split and rebalance over time.
+
+Unforunately, data from multiple sources cannot be safely aggregated in the same
+way as multiple data points from the same series can be downsampled; each series
+must be stored separately. However, Cockroach *can* optimize for this situation
+by altering the keyspace; Cockroach can store data from multiple sources
+contiguously in the key space, ensuring that the multiple series can be queried
+together more efficiently.
+
+This is done by creating a "source key", an optional identifier that is separate
+from the series name itself. Source keys are sorted *after* series name and
```

Comment:
"sorted *after*" reads awkwardly to me. I think it would be clearer to say that source keys are used as the suffix of the key or are appended to the key after the series name and timestamp.